src/lib/logical_query_plan/functional_dependency.cpp

```diff
@@ -0,0 +1,51 @@
+#include "functional_dependency.hpp"
+
+namespace opossum {
+
+FunctionalDependency::FunctionalDependency(ExpressionUnorderedSet init_determinants,
+                                           ExpressionUnorderedSet init_dependents)
+    : determinants(init_determinants), dependents(init_dependents) {
+  DebugAssert(!determinants.empty() && !dependents.empty(), "FunctionalDependency cannot be empty");
+}
+
+bool FunctionalDependency::operator==(const FunctionalDependency& other) const {
+  // Cannot use unordered_set::operator== because it ignores the custom equality function.
+  // https://stackoverflow.com/questions/36167764/can-not-compare-stdunorded-set-with-custom-keyequal
```

Comment:
bloody hell, cpp. :(

Reply:
Oh well, that's what I ran into as well. After debugging, making sure that the sets were equal, I built the `operator==` and it simply worked. 
I did not invest any energy to investigate this issue further.

And to be honest: Having read the SO-post and `std::is_permutation`, I still don't understand it. Anyway. As long as it works now. 😅